core/network/impl/synchronizer_impl.cpp

```diff
@@ -1406,4 +1408,81 @@ namespace kagome::network {
     ancestry_.clear();
     recent_requests_.clear();
   }
+
+  bool SynchronizerImpl::fetchHeadersBack(primitives::BlockNumber max, primitives::BlockNumber min,
```

Comment:
`BlockInfo` for hash check
```suggestion
  bool SynchronizerImpl::fetchHeadersBack(const primitives::BlockInfo &max, primitives::BlockNumber min, bool finalized
```

Reply:
bool finalized is regarded to what? to max block?

Reply:
Yes, should we `assignNumberToHash` or not.
`BlockTree::isFinalized` can't be used here, because we don't have these blocks in db.
We fetch parents, so they inherit finalized status.